modules/library/render/src/main/java/org/geotools/renderer/label/LabelCacheImpl.java

```diff
@@ -1265,6 +1272,43 @@ private boolean paintPointLabelInternal(LabelPainter painter, AffineTransform te
             return true;
         }
     }
+    
+    /**
+     * Splits the polygon into its component lines and labels those, one by one
+     */
```

Comment:
Sorry Andrea my bad, I misunderstood this part ``and labels those, one by one`` :disappointed: